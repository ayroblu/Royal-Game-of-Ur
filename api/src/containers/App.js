import React, { Component } from 'react'
import { withRouter, Link, Switch, Route } from 'react-router-dom'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import Home from './Home'
import Game from './Game'
import NoMatch from '../components/NoMatch'

import * as userActions from '../actions/user'
import {GraphQLApi} from '../api'

const mainQuery = `
{
  person(id:1){
    firstName
  }
}
`
class App extends Component {
  componentWillMount(){
    const api = new GraphQLApi()
    api.runQuery(mainQuery).then(res=>{
      console.log(res)
    }).catch(err=>{
      console.error('Connection error', err)
    })
  }
  render(){
    return (
      <div>
        <Switch>
          <Route exact path="/" component={Home}/>
          <Route exact path="/game/:roomId" component={Game}/>
          <Route component={NoMatch}/>
        </Switch>
        <div>
          <Link to='/game/1'>Game</Link>
        </div>
        <div>
          <Link to='/'>Home</Link>
        </div>
      </div>
    )
  }
}

export default withRouter(connect(state=>({
  user: state.user
}), dispatch=>({
  userActions: bindActionCreators(userActions, dispatch)
}))(App))
