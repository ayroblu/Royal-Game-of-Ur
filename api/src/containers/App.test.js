import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import sinon from 'sinon'
import renderer from 'react-test-renderer'

import configureStore from '../store'
import App from './App'

const initialState = {}
const store = configureStore(initialState)
store.dispatch({type: 'HOME/SET', payload: {nextGame: '123'}})

var server
beforeEach(()=>{
  server = sinon.fakeServer.create()
  server.respondWith('POST', '/api', JSON.stringify({data: {games: []}}))
})
afterAll(()=>{
  server.restore()
})
it('renders correctly', async ()=>{
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>
  ).toJSON()
  server.respond()

  //restore native XHR constructor
  await new Promise((y,n)=>setTimeout(y))
  expect(tree).toMatchSnapshot()
})
it('renders without crashing', async ()=>{
  const div = document.createElement('div');

  ReactDOM.render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>
  , div)

  server.respond()

  //restore native XHR constructor
  await new Promise((y,n)=>setTimeout(y))
  expect(server.requests.length).toEqual(1)
  expect(server.requests[0].url).toEqual('/api')
  expect(server.requests[0].status).toEqual(200)
})
