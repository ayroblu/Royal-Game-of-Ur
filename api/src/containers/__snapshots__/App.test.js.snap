exports[`test renders correctly 1`] = `
<div
  className="Home">
  <h1>
    The Royal Game of Ur
  </h1>
  <p>
    This is a 4,500 year old game, one of the oldest, if not the oldest board game in history
  </p>
  <div>
    <h2>
      Here\'s the youtube video that inspired me to make this, as well as where I get the basic rules from
    </h2>
    <p>
      I highly recommend watching this video, its very entertaining and very interesting
    </p>
    <iframe
      allowFullScreen={true}
      frameBorder="0"
      height="315"
      src="https://www.youtube.com/embed/WZskjLq040I"
      width="560" />
  </div>
  <div
    className="Left">
    <h2>
      Basic rules
    </h2>
    <p>
      The game is similar to a smaller version of ludo.
    </p>
    <p>
      You have seven pieces, you want to get your pieces on and around the track to the finish.
    </p>
    <p>
      The middle section is common ground meaning you can remove opponent pieces, or they can remove your pieces when you land on opposing pieces.
    </p>
    <p>
      There are two types of special squares, one is for a reroll and the other is where your piece cannot be removed, and an opposing piece simply moves to the next square.
    </p>
    <p>
      The dice roll is a set of 4 D4\'s, tetrahedrons, where each has 2 out of 4 white points, where if a white point is sitting up, it counts as a move. For the purposes of this version, the math is the same under the hood, but I just output the move counter and available moves.
    </p>
    <p>
      First to 7 points wins!
    </p>
  </div>
  <div
    className="start">
    <a
      href="/game/123"
      onClick={[Function]}>
      Start a game!
    </a>
  </div>
  <section
    className="existingGames">
    <h2>
      Existing Games
    </h2>
    <div
      className="games">
      <div
        className="block">
        Loading...
      </div>
    </div>
  </section>
  <div
    className="Left">
    <h2>
      Wishlist
    </h2>
    <p>
      This was a nice side project, but I\'d really like to add the following
    </p>
    <ul>
      <li>
        Better graphics and images which more accurate reflect the game board
      </li>
      <li>
        Better animations, 3D, dice rolls
      </li>
      <li>
        Game chat
      </li>
      <li>
        Game names
      </li>
      <li>
        Mobile support
      </li>
      <li>
        So much more!
      </li>
    </ul>
  </div>
  <footer>
    Built by Ben, view the code at the github link 
    <a
      href="https://github.com/ayroblu/royal-game-of-ur">
      here
    </a>
  </footer>
</div>
`;
